src/net_processing.cpp

```diff
@@ -996,6 +1016,33 @@ static bool CanServeWitnesses(const Peer& peer)
     return peer.m_their_services & NODE_WITNESS;
 }
 
+void PeerManagerImpl::UpdateChainTipSet(std::set<const CBlockIndex*>& chain_tips, const CBlockIndex* potential_new_tip)
```

Comment:
Why don't pass the peer instead of `chain_tips` here, and call `GetChainTipSetForPeerNetwork` internally?

Reply:
Done!